### ISSUE TITLE
Incorrect error response when content-length header is missing

### ISSUE THREAD

**Issue by chrismathias** [spec-first/connexion#2020]
### Description
In the validator https://github.com/spec-first/connexion/blob/main/connexion/validators/abstract.py#L126 a http status code 400 error is raised if content-length header is missing or 0 when a request body is present. The server response is then "RequestBody is required", which is confusing, as the request body does exist.

### Expected behaviour
According to https://datatracker.ietf.org/doc/html/rfc7230#section-3.3.2 the correct server response should be http status code 411 "Length Required", ideally identifying that body was present but content-length not set or 0. Ideally, the error message would indicate "content-length header missing or 0".

### Actual behaviour
Server response to missing content-length header or header with value 0 is 400 "RequestBody is required".

### Steps to reproduce
Send http POST with body but missing content-length or content-length 0.

### Additional info:

Output of the commands:

- `python --version`:  Python 3.10.12
- `pip show connexion | grep "^Version\:"`  Version: 3.1.0

**Comment by chrismathias** [comment]
Yes, we had encountered this issue in our production system. Curl does add the content-length by default. I can force it not to add the content-length by specifying  `-H 'Content-Length:` . Executed with the helloworld example:

Default CURL (will add Content-Length header): 
`curl -X 'POST' \
  'http://localhost:8080/openapi/greeting/dave' \
  -H 'accept: text/plain' \
  -H 'Content-Type: application/json' \
  -d '{}'`

Response: 
`Hello dave%`

CURL without Content-Length header (explicit removal):
`curl -X 'POST' \
  'http://localhost:8080/openapi/greeting/dave' \
  -H 'accept: text/plain' \
  -H 'Content-Type: application/json' \
  -H 'Content-Length:' \
  -d '{}'

Response:
`Invalid HTTP request received.%`

Logs of the server:
`WARNING:  Invalid HTTP request received.
BadRequestProblem(status_code=400, detail='Request body must not be empty')`

**Comment by RobbeSneyders** [comment]
Thanks for the report @chrismathias!

This is not an easy fix though, since we use the `content-length` header to determine if a request body is present without consuming the stream. The RFC you linked also mentions that a user agent SHOULD add a `content-length` header, so I'm not sure if we want to add a lot of complexity to handle this.

Another less complex way to tackle this would be to still raise a 400, but change the message to "Request body is empty or content-length is missing".

**Comment by RobbeSneyders** [comment]
If we do want to go for the more correct but more complex solution, we should probably:

1. Read the first message from the receive channel to check if there is a body
2. Check that this matches the existence of the content-length
3. Use the [`insert_messages`](https://github.com/spec-first/connexion/blob/1844a2fb3f1d09a37edfeab7a7200d52e97abbd9/connexion/validators/abstract.py#L101) method to wrap it in a new stream for further processing

**Comment by RobbeSneyders** [comment]
I'm leaning more towards the simple solution currently.

**Comment by jledrumics** [comment]
Had the same issue by upgrading to connexion v3.  
Lost a huge amount of time to understand what was the issue ahah.  
But in my opinion,  as the RFC is : 
```
A user agent SHOULD send Content-Length in a request when the method defines a meaning for enclosed content and it is not sending [Transfer-Encoding](https://httpwg.org/specs/rfc9112.html#field.transfer-encoding). For example, a user agent normally sends Content-Length in a POST request even when the value is 0 (indicating empty content). A user agent SHOULD NOT send a Content-Length header field when the request message does not contain content and the method semantics do not anticipate such data.
```
THe issue is on the user agent side

**Comment by strager** [comment]
I hit this bug today. My HTTP client (https://ogen.dev/) is using `transfer-encoding:chunked`, not `content-length:` as expected by connexion.

I think the correct behavior is to accept the request rather than merely improve the error message.
